src/common/storage/src/parquet.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2023 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_arrow::arrow::datatypes::Schema as ArrowSchema;
+use common_arrow::arrow::io::parquet::read as pread;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use opendal::Operator;
+
+pub async fn read_parquet_schema_async(operator: &Operator, path: &str) -> Result<ArrowSchema> {
```

Comment:
It seems we no need to add a new file for the `read_parquet_schema_async`

Reply:
prepare to move more utils here,  such that mainly deps on only arrow and dal (but not context)

Reply:
parquet table,  fuse, parquet copy(a simpler imply not need projection/pushdown)